src/validation.cpp

```diff
@@ -3390,6 +3453,15 @@ bool CheckBlock(const CBlock& block, CValidationState& state, const Consensus::P
     if (fCheckPOW && fCheckMerkleRoot)
         block.fChecked = true;
 
+    boost::posix_time::ptime finish = boost::posix_time::microsec_clock::local_time();
+    boost::posix_time::time_duration diff = finish - start;
+    statsClient.timing("CheckBlock_us", diff.total_microseconds(), 1.0f);
+    statsClient.gauge("blocks.currentSizeBytes", ::GetSerializeSize(block, PROTOCOL_VERSION), 1.0f);
```

Comment:
2 problems with these:
1. This is in CheckBlock, which not necessarily is for the current/last block. This block might be rejected afterwards but the metrics will still be used. All this should probably be in `ConnectBlock` instead.
2. I'm not happy with the naming of the metrics. Maybe something like `blocks.tip.sizeInBytes` and so on?

not sure I'm happy with the naming of these metrics. This is about the last processed (actually just checked, but see further down